library/ssl_tls.c

```diff
@@ -978,25 +989,25 @@ void ssl_calc_verify_ssl( mbedtls_ssl_context *ssl, unsigned char hash[36] )
     memset( pad_1, 0x36, 48 );
     memset( pad_2, 0x5C, 48 );
 
-    mbedtls_md5_update( &md5, ssl->session_negotiate->master, 48 );
-    mbedtls_md5_update( &md5, pad_1, 48 );
-    mbedtls_md5_finish( &md5, hash );
+    mbedtls_md5_update_ext( &md5, ssl->session_negotiate->master, 48 );
```

Comment:
Agree, this should be addressed separately as it is quite involved to fix. I have updated the ChangeLog and will create when the PR is merged as otherwise this does not make sense.